accepted/0019-app-notifications.md

```diff
@@ -0,0 +1,111 @@
+
+When generating an Alert, the following (bare minimum) parameters should be available:
+
+```js
+{
+  "timestamp": 1585817608, // An epoch timestamp of when the issue occurred. Generated automatically
+  "priority": "CRITICAL", // Enum of CRITICAL, WARNING, INFO // Required? Could default to warning?
+  "summary": "", // Required. HTML-anchor capable. A short summary of the issue, visible from the main window
+  "description": "", // Optional. A lengthy, more informative HTML string providing more context and inline solutions. Might support newlines and dot points
+  "persist": true, // Optional, default to false. If the user dismisses the alert, whether it should persist in the full list. Useful for alerts that are tied to a fixable issue.
+}
+```
+
+Creating this Alert should return a unique identifier, so that the plugin can keep track and control the status of an Alert. Alongside this, we should also provide an ability to tie Alerts to a specific source (scene? group?), property, and its valid state. That way the plugin relinquishes control of the Alert and it's dismissed as part of the property becoming valid, or the source being deleted/unloaded (when switching Scene collections or closing OBS).
+
+Could be designed similar to hotkeys in terms of API. Examples: `obs_add_source_notification` / `obs_add_frontend_notification` / `obs_add_output_notification`.
+
+Examples
+
+# How We Teach This
+
+On first launch, it'd be nice to show an initial bit of information via this system to introduce the user to OBS, and to explain the purpose of the Alert UI.
+
+For developers, making sure this is well documented is important. Knowing that they can reach out to the user within the UI when things go wrong is important.
+
+# Drawbacks
+
+Complexity. This will require a lot of code, logic, and a background runner keeping track of active/dismissed alerts. The solution to this may be the option to turn it off somehow for bigger productions that only need issues confirmed before going live.
+
+Visual noise. This one is less of a concern, but at the same time we don't want too much going on in the UI. A big toolbar along the top could be distracting, but at the same time going subtle means that the user could potentially not realise that something has gone wrong.
+
+# Additional Information
+
+Alerts shouldn't have a timeout, so that they're not missed by the user.
+
+There are many things to consider in this design. It's highly likely that the backend & UI would be developed & PR'd independently rather than together. This'd ensure a solid backend design, and an easier time reviewing it.
+
+The backend design & scope should be fleshed out significantly more before a real impelementation is started.
+
+# Alternatives
+
+- Exposing QSystemTrayIcon's `showMessage` - providing plugins a way to use native notifications.
+  - Downsides include:
+    - They will trigger a sound that the user cannot mute
```

Comment:
The downside of this is that the user is then disabling standard OBS notifications - like the current "Disconnected. Reconnecting.." one because they want to quiet other notifications.

On top of that, they don't visually persist, which is the upside of the proposed system.

Reply:
It sounds like you're saying that the downside is not "They will trigger a sound that the user cannot mute" but that "the user cannot customize which notifications provide a sound and which do not".

Notifications persist in the notification list up to the limit set in the notification settings (see image provided in [my comment on this PR](https://github.com/obsproject/rfcs/pull/19#issuecomment-633283181).